src/EditorFeatures/CSharp/InlineRename/CSharpEditorInlineRenameService.cs

```diff
@@ -18,4 +26,105 @@ internal sealed class CSharpEditorInlineRenameService(
     [ImportMany] IEnumerable<IRefactorNotifyService> refactorNotifyServices,
     IGlobalOptionService globalOptions) : AbstractEditorInlineRenameService(refactorNotifyServices, globalOptions)
 {
+    protected override async Task<ImmutableDictionary<string, ImmutableArray<string>>> GetRenameContextCoreAsync(IInlineRenameInfo inlineRenameInfo, IInlineRenameLocationSet inlineRenameLocationSet, CancellationToken cancellationToken)
+    {
+        var seen = PooledHashSet<TextSpan>.GetInstance();
```

Comment:
```suggestion
        using var _1 = PooledHashSet<TextSpan>.GetInstance(out var seen);
```

same for the array builders below.

Reply:
this whole method needs docs :)